Components/ConvertCSVToArray/ConvertCSVToArrayOfObjects.js

```diff
@@ -0,0 +1,60 @@
+import { convertStringToNumber } from "convert-string-to-number";
+
+export default function ConvertCSVToArrayOfObjects(
+  data,
+  { header, separator }
+) {
+  const csv = data;
+
+  const rows = csv.split("\n");
+
+  // rows contains an inpredictable number of extra rows at the end which are empty
+  // strings with unpredictable length (the function was impure), so we remove the
+  // blank spaces from the strings in rows
+  const rows2 = rows.map((row) => {
+    return row.trim();
+  });
+
+  // rows2 now contains extra rows with "", so we filter out these strings
+  const rows3 = rows2.filter((row) => {
+    return row !== "";
+  });
+
+  const array = [];
+
+  let headerRow;
+  let headerObj;
+  const content = [];
+
+  rows3.forEach((row, idx) => {
```

Comment:
I find this forEach rather hard to follow due to the naming within.

Please take another look at it and name arguments, parameters and variables in a way that makes it clear what they stand for

Reply:
We changed idx to index.